### PR TITLE
metadata: avoid to apply on hovered image instead of selection

### PR DESCRIPTION
**Pull request by phweyland**
attempt to fixe #11589

This should fix the above issue but I see a remaining problem: when a non validated field loses the focus the edited value is lost. It should be saved and restored when the focus is given back. This change is bit complicated as it implies to differentiate visually a field currently edited from a field already validated. Would the subject of another PR.

### CODE REVIEW

**Review by TurboGit — CHANGES_REQUESTED** [review]
This does not work on my side.
- I select an image
- I enter "azerty" on the title
- I move the mouse over some other pictures
- I press enter
- The title of the hovered image is changed

BTW, note that the same issue is found with the keywords.

File: src/libs/metadata.c [hunk 1]
```diff
@@ -271,6 +272,17 @@ static void _write_metadata(GtkTextView *textview, dt_lib_module_t *self)
   }
 
   GList *imgs = dt_act_on_get_images(FALSE, TRUE, FALSE);
+  // workaround: if hovered image instead of selected, aborts
+  if(imgs && !imgs->next)
+  {
+    int32_t hovered = dt_control_get_mouse_over_id();
```
Comment (TurboGit) [inline src/libs/metadata.c:278]: could be const

**Comment by phweyland** [comment]
What a shame ! At the last minute I've found dt_control_get_mouse_over_id() faster than dt_selection_get_list() and it seems I haven't tested it properly... Sorry for inconvenience.

**Comment by phweyland** [comment]
I'll have a look on tagging too.

**Comment by phweyland** [comment]
Same workaround done for tagging.

**Review by TurboGit — APPROVED** [review]
Works for me now, thanks!

**Pull request merged**